samples/oauth_rails_vethealth/app/models/oauth_callback.rb

```diff
@@ -0,0 +1,47 @@
+class OauthCallback < ApplicationRecord
+  serialize :response_body_raw
+
+  has_one :authentication
+
+  def auth(protect_secrets=true)
+    pass =
+      if protect_secrets
+        '<shhhhhhh>'
+      else
+        ENV['va_developer_client_secret']
+      end
+    { username: ENV['va_developer_client_id'], password: pass }
+  end
```

Comment:
How will the `protect_secrets == true` path be useful? Won't that always result in a failed request for the access token?

Reply:
This app took an approach to be "educational" in that using the app will walk you through what's going on behind the scenes.  So in the views it shows the `auth` information and I didn't want to expose the client secret.  But in the actual request it's set to false because we want to use the client secret.